src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -311,6 +312,15 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(schema/defn ^:always-validate valid-static-file-path?
+  "Helper function to decide if a static_file_content path is valid.
+  The access here is designed to mimic Puppet's file_content endpoint."
+  [path :- schema/Str]
+  (when-let [canonicalized-path (URIUtil/canonicalPath path)]
+    (or
+     (bidi.bidi/match-route [["modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path)
+     (bidi.bidi/match-route [["environments/" :environment-name "/modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path))))
+
```

Comment:
I've been staring at this and not making much sense out of it.  I need to grok `match-route` better, but ... can this be simplified at all and/or should there be a comment added explaining it?  Also these lines are quite long and it'd be nice to shorten them.